tidy3d/web/auth.py

```diff
@@ -73,10 +86,13 @@ def get_credentials() -> None:
             return
 
         except Exception:  # pylint:disable=broad-except
-            print("Error: Failed to log in with saved credentials.")
+            logging.info("Error: Failed to log in with saved credentials.")
 
     # keep trying to log in
-    while True:
+    attempts = 0
```

Comment:
could also just do `for attempts in range(MAX_ATTEMPTS):` I think (or `for _ in range(MAX_ATTEMPTS)` if `attempts` is not used.